src/app/catalog/components/issuer-catalog/issuer-catalog.component.ts

```diff
@@ -415,6 +424,21 @@ export class IssuerCatalogComponent extends BaseRoutableComponent implements OnI
 			},
 		};
 	}
+
+	navigateToIssuer(issuerData) {
+		if (!this.loggedIn) {
+			this.router.navigate(['/public/issuers/', issuerData.slug]);
+		} else {
+			// Check whether user is a mebemr of the institution to be redirected to `issuer/issuer` page
+			this.profileManager.userProfilePromise
+				.then((profile) => profile.emails.loadedPromise)
+				.then((emails) => {
+					const userEmail = emails.entities[0].email;
```

Comment:
To me it seems as if it would be nicer if this was a function in the backend API, but I guess it's also ok here. But are you sure that the link is only done via the first email of the user?

Reply:
Yes

Reply:
But I'm unsure if this is relevant, because I guess that the link *should* indeed happen via the primary mail

Reply:
Yes, it's relevant as the primary email isn't always the first in list. I made an update, now it uses primary email.